tests/test_functional.py

```diff
@@ -655,6 +655,23 @@ def test_obal_repoclosure_katello_with_downloaded_rpms():
     assert_in_mockbin_log(expected_log)
 
 
+@obal_cli_test(repotype='upstream')
+def test_obal_repoclosure_with_check_repo():
+    assert_obal_success(['repoclosure', 'hello', '--check', 'https://test.example.com/myrepo', '--dist', 'el7'])
+
+    expected_log = [
+        "dnf repoclosure",
+        "--config",
+        "repoclosure/yum.conf",
+        "--newest",
+        "--check downloaded_rpms",
+        "--check check_repo",
+        "--repofrompath=check_repo,https://test.example.com/myrepo",
```

Comment:
this looks weird where we tell to `--check` two repos, but only really define one of them. I know it works, but still.

Is there ever a case where we want both the remote repo and "downloaded rpms" be checked together?
Otherwise we could call both repos something like `check_repo` and only pass `--check` once?

Reply:
It is sort of ugly and tricky due to how much if/else checking and construction has to happen. For example, constructing this variable (https://github.com/theforeman/obal/pull/362/files#diff-1f4bf62d8dfa1c838d5fae334e87225b942dc26f92086eb2ab754cbe77237a24R4). Honestly, I feel like the whole role needs an overhaul and introduction of a module to handle things cleanly and do the logic more cleanly inside python code. 

My initial goal with this support was to not break the Koji method while providing a fix for the Copr method.

Reply:
In addition, this gets a big ugly too because in CI I am going to have to write the copr repository out to a file from the build step that can then be used as input to the repoclosure step.

An all-in-one `obal scratch` is nice for users, but it creates a bit of yuck in CI where we need to pass data around from invocation to invocation.

Status output has a similar issue, having to print out a yaml file so that the status generation can then slurp it in (https://github.com/theforeman/jenkins-jobs/blob/master/theforeman.org/pipelines/lib/copr.groovy#L1).

Reply:
Ehh, I decided to go ahead and write a module and then I can re-factor on top of that: https://github.com/theforeman/obal/pull/364